lib/panel/dock.js

```diff
@@ -36,6 +54,40 @@ class PanelDock {
 
     ReactDOM.render(<Component delegate={delegate} />, this.element)
   }
+  // NOTE: Chose a name that won't conflict with Dock APIs
+  doPanelResize(forConfigHeight: boolean = false) {
+    const paneContainer = getPaneContainer(this)
+    let minimumHeight: number | null = null
+    const paneContainerView = atom.views.getView(paneContainer)
+    if (paneContainerView && this.alwaysTakeMinimumSpace) {
+      // NOTE: Super horrible hack but the only possible way I could find :((
+      const dockNamesElement = paneContainerView.querySelector('.list-inline.tab-bar.inset-panel')
+      const dockNamesRects = dockNamesElement ? dockNamesElement.getClientRects()[0] : null
+      const tableElement = this.element.querySelector('table')
+      const panelRects = tableElement ? tableElement.getClientRects()[0] : null
+      if (dockNamesRects && panelRects) {
+        minimumHeight = dockNamesRects.height + panelRects.height + 1
+      }
+    }
+
+    if (paneContainer) {
+      let updateConfigHeight: number | null = null
+      const heightSet = minimumHeight !== null && !forConfigHeight ? Math.min(minimumHeight, this.panelHeight) : this.panelHeight
```

Comment:
If I understand this code correctly, it looks like you are using the config-set `panelHeight` as a maximum height for the panel in case the auto-sized height is too large. Awesome job.